### PR TITLE
Update time.js

### PR DESCRIPTION
**Pull request by mucaho**
Simpler iteration

### CODE REVIEW

**Comment by mucaho** [comment]
Added changes so that it fixes #396 
The changes do not break existing code.

**Comment by mucaho** [comment]
Easy way to test the new functionality:

``` javascript
<script type="text/javascript">
    window.onload = function() {
        //start crafty
        Crafty.init(400, 336);
        //Crafty.canvas.init();

        Crafty.e("Delay")
            .delay(function() {
                alert("Despicable me!");
                //You can also use Infinity without quotes for repeat count
            }, 3000, 2);
    };
</script>
```

File: src/time.js [hunk 1]
```diff
@@ -7,13 +7,22 @@ Crafty.c("Delay", {
...
 					item.func.call(this);
-					// remove item from array
-					this._delays.splice(index,1);
-					index--;
+					if (item.repeat > 0 ) {
+						// reschedule item
+						item.start = now;
+						item.pause = 0;
+						item.pauseBuffer = 0;
+						if (item.repeat !== Infinity)
```
Comment (starwed) [inline src/time.js]: Infinity-1 is still Infinity, so no need for a conditional here.

File: src/time.js [hunk 2]
```diff
@@ -7,13 +7,22 @@ Crafty.c("Delay", {
...
+							item.repeat--;
+					} else if (item.repeat === 0) {
```
Comment (starwed) [inline src/time.js]: If repeat is set to a negative number, it will never be removed.

**Comment by starwed** [comment]
I added a couple of line comments; it needs to be clearer what happens when repeat is set to be negative.  A common convention is for that to repeat infinitely as well.  (I think Crafty.audio.play uses that, for instance.)

**Comment by starwed** [comment]
Thanks for adding the changes mucaho!  I'll need a little bit to make sure I understand how the component work.  (I haven't looked at "Delay" before now.)

**Comment by starwed** [comment]
Ok, looks good to me.  The `if (repeat>0) {} else if (repeat<=0){}`  bit is redundant, but not in a way that makes the code harder to read, so I'll go ahead and merge.

**Pull request merged**